src/BodyParams/BodyParamsMiddleware.php

```diff
@@ -82,10 +83,17 @@ public function __invoke(ServerRequestInterface $request, ResponseInterface $res
             }
 
             // Matched! Parse and pass on to the next
-            return $next(
-                $strategy->parse($request),
-                $response
-            );
+            try {
+                return $next(
+                    $strategy->parse($request),
+                    $response
+                );
+            } catch (MalformedRequestBodyException $exception) {
```

Comment:
@michaelmoussa I think @danizord has a good point here. The FinalHandler already will convert the exception to a 400 response, and the idea I'm working on for error handling via an outer middleware layer would as well. On top of that, it would allow users to put middleware a layer above this one that could have a try/catch block for handling the error, too — which would allow for the logging, translation, and other concerns you raised.